tests/ui/cli/crate-type-flag.rs

```diff
@@ -0,0 +1,57 @@
+//! Check that `rustc`'s `--crate-type` flag accepts `--crate-type=<valid_type>` as well as the
+//! multi-value version `--crate-type=<valid_type_1>,<valid_type_2>`.
+//!
+//! This test does not try to check if the output artifacts are valid.
+
+// FIXME(#132309): add a proper `supports-crate-type` directive.
+
+// Single valid crate types should pass
+//@ revisions: lib_ rlib_ staticlib_ dylib_ cdylib_ bin_ proc_macro_
+
+//@[lib_] compile-flags: --crate-type=lib
+//@[lib_] check-pass
+
+//@[rlib_] compile-flags: --crate-type=rlib
+//@[rlib_] check-pass
+
+//@[staticlib_] compile-flags: --crate-type=staticlib
+//@[staticlib_] check-pass
+
+//@[dylib_] ignore-musl (dylibs are not supported)
+//@[dylib_] ignore-wasm (dylibs are not supported)
+//@[dylib_] compile-flags: --crate-type=dylib
+//@[dylib_] check-pass
+
+//@[cdylib_] ignore-musl (cdylibs are not supported)
+//@[cdylib_] compile-flags: --crate-type=cdylib
+//@[cdylib_] check-pass
+
+//@[bin_] compile-flags: --crate-type=bin
+//@[bin_] check-pass
+
+//@[proc_macro_] ignore-wasm (proc-macro is not supported)
+//@[proc_macro_] compile-flags: --crate-type=proc-macro
+//@[proc_macro_] check-pass
+
+//@ revisions: multivalue multivalue_combined
+
+//@[multivalue] compile-flags: --crate-type=lib,rlib,staticlib
```

Comment:
I never knew this was a valid form, just stumbed across this while reviewing #134720 and was wondering why the crate-type parser accepted a `list_list`.